File: src/test/java/com/ragin/bdd/cucumbertests/hooks/CustomDateTimeFormatter.java
package com.ragin.bdd.cucumbertests.hooks;

import com.ragin.bdd.cucumber.datetimeformat.BddCucumberDateTimeFormat;

import java.time.format.DateTimeFormatter;
import java.util.Collections;
import java.util.List;

/**
 * Example to add a date with "yyyy.MM.dd" format.
 */
public class CustomDateTimeFormatter implements BddCucumberDateTimeFormat {
    @Override
    public List<DateTimeFormatter> formatters() {
        return Collections.singletonList(
                DateTimeFormatter.ofPattern("yyyy.MM.dd")
        );
    }
}

File: src/test/kotlin/com/ragin/bdd/cucumbertests/hooks/CustomDateTimeFormatter.kt
package com.ragin.bdd.cucumbertests.hooks

import com.ragin.bdd.cucumber.datetimeformat.BddCucumberDateTimeFormat
import java.time.format.DateTimeFormatter

/**
 * Example to add a date with "yyyy.MM.dd" format.
 */
class CustomDateTimeFormatter : BddCucumberDateTimeFormat {
    override fun formatters(): List<DateTimeFormatter> {
        return listOf(
            DateTimeFormatter.ofPattern("yyyy.MM.dd")
        )
    }
}
